hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -428,6 +429,59 @@ public AbfsRestOperation createPath(final String path, final boolean isFile, fin
     return op;
   }
 
+  public AbfsRestOperation createPathBlob(final String path, final boolean isFile, final boolean overwrite,
+                                          final HashMap<String, String> metadata,
+                                          final String eTag,
+                                          TracingContext tracingContext)
+          throws AzureBlobFileSystemException {
+    final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
+    if (isFile) {
+      addCustomerProvidedKeyHeaders(requestHeaders);
+    }
+    if (!overwrite) {
+      requestHeaders.add(new AbfsHttpHeader(IF_NONE_MATCH, AbfsHttpConstants.STAR));
+    }
+    if (eTag != null && !eTag.isEmpty()) {
+      requestHeaders.add(new AbfsHttpHeader(HttpHeaderConfigurations.IF_MATCH, eTag));
+    }
+
+    final AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
+
+    String operation = SASTokenProvider.CREATE_FILE_OPERATION;
```

Comment:
lets have another enum. In future. reason, in case something changes in future for putBlob, Dev will have a way to give steps for the putBlob in their SASTokenProvider implementation.

Reply:
This cant be changed. Its a public interface.

Reply:
In case some SAS rules changes for PutBlob and get differed from that of createPath. It would require, client to expose a new enum in future. So, suggesting maybe we have different enum from now, which will give leverage to dev to accomodate different rules for the PutBlob.